mscore/timeline.cpp

```diff
@@ -878,15 +878,80 @@ Timeline::Timeline(TDockWidget* dockWidget, QWidget* parent)
+                  "7 14 2 1",
+                  "# c #000000",
+                  ". c None",
+                  "##.#...",
+                  "##.#...",
+                  "##.#...",
+                  "##.#...",
+                  "##.#...",
+                  "##.#...",
+                  "##.#...",
+                  "##.#...",
+                  "##.#...",
+                  "##.#...",
+                  "##.#...",
+                  "##.#...",
+                  "##.#...",
+                  "##.#..."
+                  };
 
-      _barlines["Start repeat"] = leftRepeatPixmap;
-      _barlines["End repeat"] = rightRepeatPixmap;
-      _barlines["Final barline"] = finalBarlinePixmap;
-      _barlines["Double barline"] = doubleBarlinePixmap;
+      static const char* heavyBarline[] = {
+                  "6 14 2 1",
+                  "# c #000000",
+                  ". c None",
+                  "..##..",
+                  "..##..",
+                  "..##..",
+                  "..##..",
+                  "..##..",
+                  "..##..",
+                  "..##..",
+                  "..##..",
+                  "..##..",
+                  "..##..",
+                  "..##..",
+                  "..##..",
+                  "..##..",
+                  "..##.."
+                  };
+      static const char* doubleHeavyBarline[] = {
+                  "7 14 2 1",
+                  "# c #000000",
+                  ". c None",
+                  ".##.##.",
+                  ".##.##.",
+                  ".##.##.",
+                  ".##.##.",
+                  ".##.##.",
+                  ".##.##.",
+                  ".##.##.",
+                  ".##.##.",
+                  ".##.##.",
+                  ".##.##.",
+                  ".##.##.",
+                  ".##.##.",
+                  ".##.##.",
+                  ".##.##."
+                  };
+
+      QPixmap* startRepeatPixmap = new QPixmap(startRepeat);
+      QPixmap* endRepeatPixmap = new QPixmap(endRepeat);
+      QPixmap* endBarlinePixmap = new QPixmap(endBarline);
+      QPixmap* doubleBarlinePixmap = new QPixmap(doubleBarline);
+      QPixmap* reverseEndBarlinePixmap = new QPixmap(reverseEndBarline);
+      QPixmap* heavyBarlinePixmap = new QPixmap(heavyBarline);
+      QPixmap* doubleHeavyBarlinePixmap = new QPixmap(doubleHeavyBarline);
+
+      _barlines[BarLine::userTypeName(BarLineType::START_REPEAT)] = startRepeatPixmap;
+      _barlines[BarLine::userTypeName(BarLineType::END_REPEAT)] = endRepeatPixmap;
+      _barlines[BarLine::userTypeName(BarLineType::END)] = endBarlinePixmap;
+      _barlines[BarLine::userTypeName(BarLineType::DOUBLE)] = doubleBarlinePixmap;
+      _barlines[BarLine::userTypeName(BarLineType::REVERSE_END)] = reverseEndBarlinePixmap;
+      _barlines[BarLine::userTypeName(BarLineType::HEAVY)] = heavyBarlinePixmap;
+      _barlines[BarLine::userTypeName(BarLineType::DOUBLE_HEAVY)] = doubleHeavyBarlinePixmap;
```

Comment:
Maybe we could, or even should, better be using the SMuFL qlyphs instead of these bitmaps?
We do for Segno and Coda.

Reply:
That'd be great! I'm curious to see what the standard fonts have as smufl glyphs for barlines.

ha funny;) I was having a look at the smufl spcification, and happened to fall on this page : https://w3c.github.io/smufl/gitbook/specification/glyphnames.html.

Anyway, [here's the barlines page](https://w3c.github.io/smufl/gitbook/tables/barlines.html).

It might be difficult to edit the barline thickness if we use smufl glyphs, so we'd probably have to keep the pixmaps for backward comp.

Reply:
Also, the bottom note says: 
## Implementation notes

Scoring programs should draw their own barlines using primitives, not use the glyphs in this range.

Reply:
Yes, for the barlines in the staves, but here thy are for the timeline, so the glyphs should be as fine as they can get